### PR TITLE
Another attempt to fix exceptions

### PR DESCRIPTION
**Pull request by jigar-f**
This PR fixes an issue with the OutOfMemoryError exception with a different approach. Now we are writing a file with the apps list and reading it on GO end. That should fix this issue 100%

### CODE REVIEW

File: android/app/src/main/kotlin/org/getlantern/lantern/plausible/PlausibleClient.kt [hunk 1]
```diff
@@ -195,17 +195,22 @@ internal class NetworkFirstPlausibleClient(
...
-            Proxy(
-                Proxy.Type.HTTP,
-                InetSocketAddress(
-                    "127.0.0.1",
-                    uri.port,
-                ),
-            )
-        OkHttpClient.Builder().proxy(proxy).build()
+        if (hTTPAddr.isEmpty()) {
+            OkHttpClient()
```
Comment (jigar-f) [inline android/app/src/main/kotlin/org/getlantern/lantern/plausible/PlausibleClient.kt:199]: If we don't have proxyAddr lets not crash and give it an empty client.

File: android/app/src/main/kotlin/io/lantern/model/SessionModel.kt [hunk 2]
```diff
@@ -396,28 +401,53 @@ class SessionModel internal constructor(
...
+                val file = File(activity.cacheDir, "appsData.json")
+
+
+                // Write JSON data directly to the file to reduce memory usage
+                file.outputStream().use { outputStream ->
+                    Json.encodeToStream(buildJsonArray {
+                        appsList.forEach { app ->
+                            add(
+                                buildJsonObject {
+                                    val byte = app.icon ?: ByteArray(0)  // Safeguard for null icons
```
Comment (atavism) [inline android/app/src/main/kotlin/io/lantern/model/SessionModel.kt:413]: I think adding this safeguard is useful. What will the user see when an icon is missing? Blank space? Do we want to display a generic app icon as a placeholder?

File: android/app/src/main/kotlin/io/lantern/model/SessionModel.kt [hunk 3]
```diff
@@ -396,28 +401,53 @@ class SessionModel internal constructor(
...
+                                        byte.toUByteArray().forEach { add(it.toInt()) }  // Store each UByte as an Int
+                                    }
                                 }
-                            }
-                        )
-                    }
+                            )
+                        }
+                    }, outputStream)
                 }
```
Comment (atavism) [inline android/app/src/main/kotlin/io/lantern/model/SessionModel.kt:423]: This looks like a great way to mitigate some of the memory issues! I think we could optimize further by implementing some caching here as well? For instance, you could serve the JSON file from cache if it already exists and add logic to refresh it only if it’s older than 24 hours. Additionally, you could minimize file writes by re-generating the file only if the installed apps differs from the last snapshot? We could always handle this in a subsequent PR but I figured I would mention it!

File: android/app/src/main/kotlin/io/lantern/model/SessionModel.kt (hunk 2 above)
Comment (jigar-f) [reply android/app/src/main/kotlin/io/lantern/model/SessionModel.kt:413]: I don't think that is possible, Most of them have an app Icon, Anyway, I am going to add some placeholder on the Flutter end.

File: android/app/src/main/kotlin/io/lantern/model/SessionModel.kt (hunk 3 above)
Comment (jigar-f) [reply android/app/src/main/kotlin/io/lantern/model/SessionModel.kt:423]: Thanks, Looking into it.

Comment (jigar-f) [reply android/app/src/main/kotlin/io/lantern/model/SessionModel.kt:429]: Changes are done.

**Comment by jigar-f** [comment]
@atavism All changes are done, Do take look

**Comment by jigar-f** [comment]
@atavism I think we should merge this and put out a new release.

**Comment by atavism** [comment]
@jigar-f I'm testing the changes now.

**Comment by atavism** [comment]
Working well for me

**Pull request merged**